crates/services/upgradable-executor/src/storage_access_recorder.rs

```diff
@@ -0,0 +1,49 @@
+use fuel_core_storage::{
+    kv_store::{
+        KeyValueInspect,
+        StorageColumn,
+        Value,
+    },
+    Result as StorageResult,
+};
+use fuel_core_types::services::executor::StorageReadReplayEvent;
+use parking_lot::Mutex;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct StorageAccessRecorder<S>
+where
+    S: KeyValueInspect,
+{
+    pub storage: S,
+    pub record: Arc<Mutex<Vec<StorageReadReplayEvent>>>,
+}
+
+impl<S> StorageAccessRecorder<S>
+where
+    S: KeyValueInspect,
+{
+    pub fn new(storage: S) -> Self {
+        Self {
+            storage,
+            record: Default::default(),
+        }
+    }
+}
+
+impl<S> KeyValueInspect for StorageAccessRecorder<S>
+where
+    S: KeyValueInspect,
+{
+    type Column = S::Column;
+
+    fn get(&self, key: &[u8], column: Self::Column) -> StorageResult<Option<Value>> {
+        let value = self.storage.get(key, column)?;
+        self.record.lock().push(StorageReadReplayEvent {
+            column: column.name(),
```

Comment:
You can use `column.id()`(`u32`) instead of the string

Reply:
I think the column names are better for forwards-compatibility, whatever little of that we can have. Do you think the performance matters so much that it makes sense to use u32 here?

Reply:
For backward compatibility we will continue to use the same column, because STF deepens on them. So I think it is better to rely on `u32` in the same way as WASM executor.

Plus, it is less data to serialize.